File: patch_dist_latex.pl
#!/usr/bin/perl
use strict;
use warnings;

my $file = $ARGV[0] || 'dist/assets/index-DmUJCYoB.js';
open(my $fh, '<', $file) or die "Cannot read $file: $!";
my $content = do { local $/; <$fh> };
close($fh);

my $original = $content;

# Replace four backslashes with two inside latex:"..." string literals.
# In Perl source here, the regex /\\\\/ matches two literal backslashes,
# and the replacement \\ produces one literal backslash. We want to turn
# the bundle's "\\\\frac" (four backslashes, runtime value "\\frac")
# into "\\frac" (two backslashes, runtime value "\frac").
$content =~ s{(latex:")((?:[^"\\]|\\.)*)(")}{
    my ($pre, $body, $post) = ($1, $2, $3);
    $body =~ s/\\\\/\\/g;
    $pre . $body . $post
}ge;

if ($content eq $original) {
    print "No changes made.\n";
    exit 0;
}

open(my $out, '>', $file) or die "Cannot write $file: $!";
print $out $content;
close($out);

print "Patched $file\n";

File: patch_dist_latex.py
import re
import sys

def patch_latex_strings(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    original = content

    def fix_latex_string(match):
        prefix = match.group(1)  # latex:"
        body = match.group(2)    # string content
        suffix = match.group(3)  # "
        # Replace four backslashes with two inside the latex string literal
        fixed_body = body.replace('\\\\\\\\', '\\\\')
        return prefix + fixed_body + suffix

    # Match latex:"..." where ... does not contain an unescaped double quote
    # In JS string literals, quotes inside are escaped as \"
    pattern = r'(latex:")((?:[^"\\]|\\.)*)(")'
    content = re.sub(pattern, fix_latex_string, content)

    if content == original:
        print('No changes made.')
        return

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

    print(f'Patched {filepath}')

if __name__ == '__main__':
    patch_dist_latex_strings(sys.argv[1] if len(sys.argv) > 1 else 'dist/assets/index-DmUJCYoB.js')
